tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -241,7 +242,15 @@ def emit(self, record):
             'ROSCONSOLE_FORMAT', '[${severity}] [${time}]: ${message}')
         msg = msg.replace('${severity}', level)
         msg = msg.replace('${message}', str(record_message))
-        msg = msg.replace('${walltime}', '%f' % time.time())
+
+        # walltime tag
+        msg = msg.replace('${walltime}', '%f' % time.time())  # for performance reasons
+
+        while '${walltime:' in msg:
+            time_format = msg[msg.index('${walltime:') + len('${walltime:'): msg.index('}')]
```

Comment:
What if another placeholder which hasn't been replaced yet precedes the `${walltime:`? `msg.index('}')` would return an index which is *before* `msg.index('${walltime:')`.

Reply:
Ok, you're right. Line:
```
time_format = msg[msg.index('${walltime:') + len('${walltime:'): msg.index('}')]
```
should looks like
```
tag_end_index = msg.index('${walltime:') + len('${walltime:')
time_format = msg[tag_end_index: msg.index('}', tag_end_index)]
```
I'll fix it by next week.

Reply:
@dirk-thomas  I added unit test for formatted timestamp, but there are some problems on **Npr_*** builds. Could you tell me what does **Npr_** stands for?

Reply:
These are the **PR** builds for the upcoming ROS **Noetic** distribution.

Just recently the default branch of this repo has moved to `noetic-devel` so please retarget that branch instead (and then the already existing **Mpr** results won't be updated anymore).

Reply:
Is there a problem with a Jenkins pipeline script on line `git rev-parse origin/noetic-devel^{commit}` ?

```
12:00:50  > git rev-parse origin/noetic-devel^{commit} # timeout=10
12:00:50 FATAL: Command "git rev-parse origin/noetic-devel^{commit}" returned status code 128:
12:00:50 stdout: origin/noetic-devel^{commit}
12:00:50 
12:00:50 stderr: fatal: ambiguous argument 'origin/noetic-devel^{commit}': unknown revision or path not in the working tree.
```
[[Npr__ros_comm__ubuntu_focal_amd64/138/](http://build.ros.org/job/Npr__ros_comm__ubuntu_focal_amd64/138/)]

I even tried an another PR-draft on https://github.com/ros/ros_comm/pull/1892, made the same error: `FATAL: Command "git rev-parse origin/noetic-devel^{commit}" returned status code 128:` [[Npr__ros_comm__ubuntu_focal_amd64/139](http://build.ros.org/job/Npr__ros_comm__ubuntu_focal_amd64/139/)]